### PR TITLE
feat: Cart Empty Message component

### PR DESCRIPTION
**Pull request by kieckhafer**
This PR adds a `CartEmptyMessage` component.

It will render a Cart Empty message, with a button to `Continue Shopping`, when a user is viewing their cart with nothing in it.

Issue in the starterkit: https://github.com/reactioncommerce/reaction-next-starterkit/issues/138

### CODE REVIEW

File: package/src/components/CartEmptyMessage/v1/CartEmptyMessage.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  render() {
+    const { onClick } = this.props;
+    const { ContinueShoppingButton } = this.props.components;
+
+    return (
+      <Fragment>
+        <EmptyMessage>Your shopping cart is empty.</EmptyMessage>
+        <EmptyButton>
+          <ContinueShoppingButton className="emptyCartBtn" actionType="important" onClick={onClick}>Continue shopping</ContinueShoppingButton>
```
Comment (willopez) [inline package/src/components/CartEmptyMessage/v1/CartEmptyMessage.js]: class `emptyCartBtn` is not used,  remove.

Comment (willopez) [inline package/src/components/CartEmptyMessage/v1/CartEmptyMessage.js]: The button text should be a prop, and default it to "Continue shopping"

File: package/src/components/CartEmptyMessage/v1/CartEmptyMessage.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    components: PropTypes.shape({
+      ContinueShoppingButton: PropTypes.oneOfType([PropTypes.string, PropTypes.func])
+    }),
+    /**
+     * Onclick function to pass to the Button component. Not handled internally, directly passed
+     */
+    onClick: PropTypes.func
+  }
+
+  static defaultProps = {
```
Comment (willopez) [inline package/src/components/CartEmptyMessage/v1/CartEmptyMessage.js]: this is not necessary

File: package/src/components/CartEmptyMessage/v1/CartEmptyMessage.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    onClick() {}
+  };
+
+  render() {
+    const { onClick } = this.props;
+    const { ContinueShoppingButton } = this.props.components;
+
+    return (
+      <Fragment>
+        <EmptyMessage>Your shopping cart is empty.</EmptyMessage>
```
Comment (willopez) [inline package/src/components/CartEmptyMessage/v1/CartEmptyMessage.js]: User's might want to provide a custom message, make this a prop and default it to "Your shopping cart is empty."

File: package/src/components/CartEmptyMessage/v1/CartEmptyMessage.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    /**
+     * Onclick function to pass to the Button component. Not handled internally, directly passed
+     */
+    onClick: PropTypes.func
+  }
+
+  static defaultProps = {
+    onClick() {}
+  };
+
```
Comment (willopez) [inline package/src/components/CartEmptyMessage/v1/CartEmptyMessage.js]: add a function to elevate the onClick handler to the parent, like so
```
handleOnClick = () => {
  this.props.onClick()
}
```

File: package/src/components/CartEmptyMessage/v1/CartEmptyMessage.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+### CartEmptyMessage
+
+#### Overview
+The `CartEmptyMessage` displays when viewing an empty shopping cart.
+
+#### A cart emtpy message will be rendered when there are no items in your cart
```
Comment (willopez) [inline package/src/components/CartEmptyMessage/v1/CartEmptyMessage.md]: Add other cases, where custom props for button text and message are provided.

**Comment by kieckhafer** [comment]
@willopez addressed your notes, ready for another look

**Pull request merged**